client/js/screens/SignUp/SignUp.js

```diff
@@ -1,9 +1,104 @@
-import React from "react";
-import { Text } from "react-native";
+import React, { Fragment } from "react";
+import {
+  Text,
+  ImageBackground,
+  View,
+  TextInput,
+  TouchableOpacity
+} from "react-native";
 import styles from "./styles";
+import { Form, Field } from "react-final-form";
 
-const Signup = () => {
-  return <Text>I am the Signup Screen</Text>;
+const onSubmit = async (values, createUser) => {
+  const res = await createUser({ variables: values });
+};
+
+const Signup = ({ data }) => {
+  return (
+    <Fragment>
+      <ImageBackground
+        source={require("../../assets/images/Signup.png")}
+        style={styles.backgroundImage}
+      >
+        <Text style={styles.register}>REGISTER</Text>
+        <Form
+          onSubmit={values => onSubmit(values, createUser)}
+          render={({ handleSubmit, pristine, invalid }) => (
+            <View>
+              <Field name="firstname">
+                {({ input, meta }) => (
+                  <View>
+                    <TextInput
+                      {...input}
+                      style={styles.input}
+                      placeholder="First name"
+                      placeholderTextColor="white"
+                    />
+                  </View>
```

Comment:
What if there is an error? Please show the errors and validate your input.

Reply:
done

Reply:
done